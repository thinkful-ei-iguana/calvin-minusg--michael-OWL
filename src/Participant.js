import React from "react";

export default function Participant(prop) {
  return (
    <div className="session-participant" key={prop.id}>
      <img src={prop.avatar} alt={prop.name} className="user-avatar" />
      <div className="participant-info">
        <h6 className="participant-name">{prop.name}</h6>
        <div className={prop.onStage}></div>
        <div className={prop.inSession}></div>
      </div>
    </div>
  );
}
