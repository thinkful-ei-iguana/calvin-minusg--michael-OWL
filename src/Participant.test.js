import React from "react";
import ReactDOM from "react-dom";
import Participant from "./Participant";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Participant />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer
    .create(
      <Participant
        name="Calvin Rosehart"
        onStage="true"
        inSession="true"
        avatar="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
