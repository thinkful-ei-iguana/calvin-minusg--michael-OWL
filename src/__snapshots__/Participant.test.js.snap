// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="session-participant"
>
  <img
    alt="Calvin Rosehart"
    className="user-avatar"
    src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
  />
  <div
    className="participant-info"
  >
    <h6
      className="participant-name"
    >
      Calvin Rosehart
    </h6>
    <div
      className="true"
    />
    <div
      className="true"
    />
  </div>
</div>
`;
